### ISSUE TITLE
can I encapsulate opendds client in to a dll?

### ISSUE THREAD

**Issue by tantan1** [OpenDDS/OpenDDS#4283]
I can run opendds demo, but when I try to encapsulate the TheParticipantFactoryWithArgs , create_participant into a dll ddsInitDll and invoke the ddsInitDll, it throw a error.

**Comment by iguessthislldo** [comment]
Yes, OpenDDS can run from a shared library like a DLL. All the language binding we have like Java and Python basically require that.

What is the error? If you can please also post a stack trace.

Also if you're trying to do this from a DLL, you might need to call `ACE::init()` before trying to start OpenDDS. I don't think we call it automatically. I thought we had another user with a similar issue, but I couldn't find it.
